### ISSUE TITLE
Phi-3 support

### ISSUE THREAD

**Issue by Theodotus1243** [OpenNMT/CTranslate2#1672]
Powerful model trained on syntetic data, has high MMLU

4K context window one should be easier, as has no `LongRope`

https://huggingface.co/microsoft/Phi-3-mini-4k-instruct
https://arxiv.org/pdf/2404.14219.pdf

**Comment by BBC-Esq** [comment]
I second this.  The current phi loader is broken, apparently because of some changes that Microsoft did to the model after it was initially released.   At any rate, adapting the phi loader to the new phi3 should be easier than starting from scratch.

**Comment by jncraton** [comment]
For anyone else researching this, [phi3 support has been added](https://github.com/ggerganov/llama.cpp/pull/6852/files#diff-ecca4c14f9a354b5557247cafd79409b332bfa1e9c12594f83282af1fde4743e) to the convert_hf_to_gguf.py script in llama.cpp. Perhaps something can be gleaned from there to simplify the implementation of the ct2 converter.

**Comment by vince62s** [comment]
no worry it will be done, it's quite easy for the mini-4k since it takes all llama2 arch.
fyi: https://forum.opennmt.net/t/phi-3-3-8b-llama2-7b-ensemble-just-for-fun/5729

**Comment by BBC-Esq** [comment]
Is it done yet?  I've been waiting patiently for approximately two hours now?  ;-)

**Comment by minhthuc2502** [comment]
Hello, I am working on it. Some unexpected problems appears.

**Comment by BBC-Esq** [comment]
I'm not skilled enough to help directly by implementing the code...but if you want me to do any grunt work or research let me know dude...anything to assist speed up the process.  Thanks!

**Comment by BBC-Esq** [comment]
I'd like to start learning to eventually possibly help...Question...how do I get the actual model architecture to start with...It's my understanding that getting the model's structure, what activation functions are used, etc. and basically starting to understanding the structure is key in making additional converters down the road.  For example, here's a link:

[https://bbycroft.net/llm](url)

Here are some other links that I've been gathering with the goal of eventually contributing a converter...based on first trying to understand the structure of LLMs...

[https://github.com/mert-kurttutan/torchview](url)

[https://github.com/lutzroeder/netron](url)

Huggingface sometimes (but not always) has information like this...

![image](https://github.com/OpenNMT/CTranslate2/assets/108230321/5478d1d6-faae-4d0e-b0b6-7f997546d60e)

Basically, any good starting point for me that you'd recommend dude?  Thanks!

**Comment by minhthuc2502** [comment]
PR #1680 to add the converter for Phi3
